### PR TITLE
{Packaging} Optimize Linux package and docker image by trimming SDK

### PR DESCRIPTION
**Pull request by bebound**
**Description**

We have used `trim_sdk.py` and only keep pyc file on Windows. Apply them on Linux and docker to reduce package size.

This is the first part, use trim_sdk on Linux.
Using pyc is https://github.com/Azure/azure-cli/pull/25801

Result:
Ubuntu 22.04 installed size: 965MB -> 576 MB, become 60% of original size. 
RHEL 9 installed size 771MB -> 403 MB, become 52% of original size. 
Docker  1328MB -> 952MB,  become 72% of original size. 

**Testing Guide**
 

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Packaging

File: scripts/release/rpm/azure-cli.spec [hunk 1]
```diff
@@ -52,6 +52,9 @@ source %{repo_path}/scripts/install_full.sh
 # cffi 1.15.0 doesn't work with RPM: https://foss.heptapod.net/pypy/cffi/-/issues/513
 %{python_cmd} -m pip install cffi==1.14.6
 
+# Remove unused SDK version
+%{python_cmd} %{repo_path}/scripts/trim_sdk.py
```
Comment (jiasli) [inline scripts/release/rpm/azure-cli.spec:56]: How about we move this line after L50 to align with `Dockerfile`?

**Review by jiasli — APPROVED** [review]

**Comment by jiasli** [comment]
Nice work! This certainly improves #7387 a lot!

**Comment by yonzhan** [comment]
This is a great achievement.

**Pull request merged**